pkg/util/azureclient/openshiftmanagedcluster/admin/client.go

```diff
@@ -667,3 +734,31 @@ func (future *OpenShiftManagedClustersRotateSecretsFuture) Result(client OpenShi
 	ar.Response = future.Response()
 	return
 }
+
+// OpenShiftManagedClustersGetControlPlanePodsFuture an abstraction for monitoring and retrieving the results of a
+// cluster status operation.
+type OpenShiftManagedClustersGetControlPlanePodsFuture struct {
+	azure.Future
+}
+
+func (future *OpenShiftManagedClustersGetControlPlanePodsFuture) Result(client OpenShiftManagedClustersClient) (ar autorest.Response, err error) {
+	var done bool
+	done, err = future.Done(client)
+	if err != nil {
+		err = autorest.NewErrorWithError(err, "containerservice.OpenShiftManagedClustersGetControlPlanePodsFuture", "Result", future.Response(), "Polling failure")
+		return
+	}
+	if !done {
+		err = azure.NewAsyncOpIncompleteError("containerservice.OpenShiftManagedClustersGetControlPlanePodsFuture")
+		return
+	}
+	ar.Response = future.Response()
+	return
+}
+
+// OpenShiftManagedClustersControlPlanePods contains the status of the control plane pods
+type OpenShiftManagedClustersControlPlanePods struct {
+	autorest.Response `json:"-"`
+
+	Items []byte `json:"-"`
```

Comment:
Why not have a list of pods and hide the complexity of having to unmarshal from clients?

Reply:
I originally designed it the way you suggested. But @jim-minter argues that it wasn't nice to have a dependency on `v1.Pod` from the azure client.

Reply:
Well, implicitly you already have a dependency on `v1.Pod`.